### PR TITLE
Document OpenSearch 2.0 Java compatibility.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Took a stab at cleaning up the compatibility page and documenting 2.0 as part of https://github.com/opensearch-project/opensearch-plugins/issues/132.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/install/compatibility.md [hunk 1]
```diff
@@ -5,15 +5,20 @@ parent: Install OpenSearch
...
+# Java Compatibility
 
-OpenSearch version | Compatible Java versions | Recommended operating systems
-:--- | :--- | :---
-1.0 - 1.2.x | 11, 15 | Red Hat Enterprise Linux 7, 8; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04
-1.3.x | 8, 11, 14 | Red Hat Enterprise Linux 7, 8; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04
\ No newline at end of file
+The OpenSearch ditribution for Linux ships with a compatible [Adoptium JDK](https://adoptium.net/) version of Java in the `jdk` directory. To find its version, run `./jdk/bin/java -version`. For example, the OpenSearch 1.0.0 tarball ships with Java 15.0.1+9 (non-LTS), OpenSearch 1.3.0 includes Java 11.0.14.1+1 (LTS), and OpenSearch 2.0.0 includes Java 17.0.2+8 (LTS).
+
+OpenSearch 1.0 to 1.2.4 is built and tested with Java 15, OpenSearch 1.3.0 is built and tested with Java 8, 11 and 14, and OpenSearch 2.0.0 is built and tested with Java 11 and 17.
```
Comment (Naarcha-AWS) [inline _opensearch/install/compatibility.md:16]: Does the matrix below already clarify compatible and tested versions?

If we want to keep this paragraph, lets make it three separate sentences.

Comment (dblock) [reply _opensearch/install/compatibility.md:16]: I deleted it and simplified the sentence.

File: _opensearch/install/compatibility.md [hunk 2]
```diff
@@ -5,15 +5,18 @@ parent: Install OpenSearch
...
-1.0 - 1.2.x | 11, 15 | Red Hat Enterprise Linux 7, 8; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04
-1.3.x | 8, 11, 14 | Red Hat Enterprise Linux 7, 8; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04
\ No newline at end of file
+The OpenSearch distribution for Linux ships with a compatible [Adoptium JDK](https://adoptium.net/) version of Java in the `jdk` directory. To find the JDK version, run `./jdk/bin/java -version`. For example, the OpenSearch 1.0.0 tarball ships with Java 15.0.1+9 (non-LTS), OpenSearch 1.3.0 ships with Java 11.0.14.1+1 (LTS), and OpenSearch 2.0.0 ships with Java 17.0.2+8 (LTS). OpenSearch is tested with all compatible Java versions.
+
+OpenSearch Version | Compatible Java Versions | Bundled Java Version
+:---------- | :-------- | :-----------
+1.0 - 1.2.x | 11, 15    | 15.0.1+9
+1.3.x       | 8, 11, 14 | 11.0.14.1+1
+2.0.0       | 11, 17    | 17.0.2+8
```
[the remark this answers is not in the document]
Comment (dblock) [reply _opensearch/install/compatibility.md:20]: The exact version bundled is the third column. What am I missing?

**Comment by dblock** [comment]
@dlvenable Can we get this merged?

**Review by dlvenable — APPROVED** [review]

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Comment by Naarcha-AWS** [comment]
@dblock: I'm going to take over this PR to ensure gets merged in for 2.0.0

**Comment by Naarcha-AWS** [comment]
Closing in favor of #609

**Pull request closed without merging**